xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java

```diff
@@ -334,9 +371,15 @@ public void reportRecordedLoadData() {
             .setDroppedCount(0))
         .setTotalDroppedRequests(0)
         .build();
+
+    // Add load stats source for some cluster service.
     when(loadStatsStore.generateLoadReport()).thenReturn(expectedStats1, expectedStats2);
+    lrsClient.addLoadStatsStore("namespace-foo:service-blade", loadStatsStore);
+
+    InOrder inOrder = inOrder(requestObserver, loadStatsStore);
```

Comment:
Why do you include `loadStatsStore` into `inOrder`? You are not verifying anything over `loadStatsStore` since then.

Reply:
I see, it's actually used in `assertNextReport()`.

Reply:
In the helper method `assertNextReport`, it does verify `loadStatsStore`. I changed `assertNextReport` signature to take in the `LoadStatsStore` instance to be orderly verified with the `requestObserver`.

`LoadReportClientImplTest.java` contains some badly written tests (at line 227, I put a note about this) and some bad programming practices (or even something incorrect) I did not realize until recently. I am trying to fix (possibly not all of them) those that look significantly bad.